mono/metadata/threads-types.h

```diff
@@ -117,7 +119,7 @@ ICALL_EXPORT
 int ves_icall_System_Threading_Thread_GetPriority (MonoThreadObjectHandle this_obj, MonoError *error);
 
 ICALL_EXPORT
-void ves_icall_System_Threading_Thread_SetPriority (MonoThreadObjectHandle this_obj, int priority, MonoError *error);
+void ves_icall_System_Threading_Thread_SetPriority (MonoThreadObjectHandle this_obj, MonoThreadPriority priority, MonoError *error);
```

Comment:
You have no guarantee on the size of the enum in Native, while managed will always pass an int. Revert that change.